### ISSUE TITLE
DateTime.seconds being Decimal is incompatible with builtin datetime

### ISSUE THREAD

**Issue by Joshuaalbert** [neo4j/neo4j-python-driver#773]
When converting `neo4j.time.DateTime` to `datetime.datetime` we find that the `seconds` in neo4j is unexpectedly a decimal, whereas all other fields are integer (as expected). Therefore, in python 3.10 where this casting depreceated we get an error.
```bash
TypeError: 'decimal.Decimal' object cannot be interpreted as an integer
```

# minimal example
This fails on python 3.10+

```python
from neo4j.time import DateTime
from datetime import datetime
neo4j_dt = DateTime(2022,1,1)

# this fails because neo4j_dt.second is Decimal type
datetime(year=neo4j_dt.year, month=neo4j_dt.month, day=neo4j_dt.day, hour=neo4j_dt.hour,
                     minute=neo4j_dt.minute, second=neo4j_dt.second, microsecond=neo4j_dt.nanosecond // 1000,
                    tzinfo=neo4j_dt.tzinfo)

print(neo4j_dt.second)
# Decimal('0E-9')

# setting seconds to zero reveal that it's the only DateTime property that is a Decimal
datetime(year=neo4j_dt.year, month=neo4j_dt.month, day=neo4j_dt.day, hour=neo4j_dt.hour,
                     minute=neo4j_dt.minute, second=0, microsecond=neo4j_dt.nanosecond // 1000,
                    tzinfo=neo4j_dt.tzinfo)
```

**Comment by robsdedude** [comment]
Hello and thanks for reaching out.

Sorry for the long wait. I was hoping I could come back to you with good news, but we didn't find the time yet. So what's going on here is, as you observed correctly, that the 4.4 driver is not compatible with Python 3.10. If you have a look at the [API docs](https://neo4j.com/docs/api/python-driver/4.4/) you'll see that this is also documented. I'm pretty positive there are other things that are also broken when using Python 3.10 with the 4.4 driver. It's definitely untested.

As alluded to before, I was hoping to be able to come back to you with good news, i.e., added Python 3.10 support for 4.4. But it does not have a very high priority right now nor can I guarantee that it will ever be shipped. It might. In the meantime, you might be happy to hear that the 5.0 driver that just released today fully supports Python 3.10.

**Comment by robsdedude** [comment]
Giving this a seconds look lets me think that this isn't even a driver bug.

It's simply how the neo4j DateTime API works in 4.4 ([API docs](https://neo4j.com/docs/api/python-driver/4.4/temporal_types.html#neo4j.time.DateTime.second)). Note: in 5.0 `neo4j.DateTime().second` returns an `int` and your code would work).

However, you should probably just use `neo4j.DateTime().to_native()`. That way you don't even have to bother with any type conversions.

Closing this as this particular error is not a driver bug. But please feel free to keep commenting if you have more questions.
